internal/cmd/helm-operator/run/cmd.go

```diff
@@ -179,18 +177,11 @@ func run(cmd *cobra.Command, f *flags.Flags) {
 
 	for _, w := range ws {
 
-		// TODO: remove this after modifying watches of hybrid lib.
-		cl, err := getChart(w)
-		if err != nil {
-			log.Error(err, "Unable to read chart")
-			os.Exit(1)
-		}
-
 		r, err := reconciler.New(
-			reconciler.WithChart(*cl),
+			reconciler.WithChart(*w.Chart),
 			reconciler.WithGroupVersionKind(w.GroupVersionKind),
 			reconciler.WithOverrideValues(w.OverrideValues),
-			reconciler.WithSelector(w.Selector),
+			reconciler.WithSelector(*w.Selector),
```

Comment:
if w.Selector is nil does this fail with a nil pointer exception?

Reply:
since its optional, nope. If selector is nil, we won't have predicate created to filter resources https://github.com/operator-framework/helm-operator-plugins/blob/98d5f4f1188e1c1fa8aa7695cd287d1f403e5fdf/pkg/reconciler/reconciler.go#L865